src/nvim/api/extmark.c

```diff
@@ -407,6 +407,17 @@ Array nvim_buf_get_extmarks(Buffer buffer, Integer ns_id, Object start, Object e
 ///               - strict: boolean that indicates extmark should not be placed
 ///                   if the line or column value is past the end of the
 ///                   buffer or end of the line respectively. Defaults to true.
+///               - sign_text: string of length 1-2 used to display in the
+///                   sign column.
+///               - sign_hl_group: name of the highlight group used to
+///                   highlight the sign column text.
+///               - number_hl_group: name of the highlight group used to
+///                   highlight the number column.
+///               - line_hl_group: name of the highlight group used to
```

Comment:
I didn't initially plan to add `sign_hl_group` as I thought `hl_goup` could be used instead, however I've found there's no way to highlight the entire line using `hl_group`.

Reply:
how does it differ from using `hl_eol` ?

Reply:
`hl_eol` requires `end_row` to be provided and to be at least 1 greater than `start_row`. This makes it tricky to be used with signs in a single extmark since you don't want the `end_row` to have a sign meaning you would need to use 2 different extmarks.

Reply:
ok, so it makes sense from a sign feature compat perspective. let's keep it then.